sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -12,14 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import re
-from collections import namedtuple
-from typing import List, Dict, Union
-
-# TODO: Move this to a separate class
-# For now, this identifies a condition with only "==" operator supported.
-ConditionOperator = namedtuple('ConditionOperator',
-                               'operator operand1 operand2')
-PipelineParamTuple = namedtuple('PipelineParamTuple', 'name op pattern')
+import attr
+from typing import List, Dict, Union, Any
+import warnings
+
+@attr.s
+class ConditionOperator:
+  operator = attr.ib(type=str)
+  operand1 = attr.ib(type=Union['PipelineParam', Any])
+  operand2 = attr.ib(type=Union['PipelineParam', Any])
+
+  def __bool__(self) -> bool:
+    msg = ' '.join([
+      "Treating comparison of PipelineParam as bool,",
+      "but its value is not known until the pipeline is run.",
+      "If operator combinator was meant, use & or | instead."
+    ])
+    warnings.warn(msg)
+    return super().__bool__()
+
+  def __and__(self, other) -> 'ConditionOperator':
+    return ConditionOperator('&&', self, other)
+
+  def __or__(self, other) -> 'ConditionOperator':
```

Comment:
Can we support `not`?

Reply:
Sure, it would require either:
* ignoring one of the arguments (probably setting it to `None`?), like postfix operators in C++
* creating a separate class `UnaryOperator` (and rename this one to `BinaryOperator`)
* changing `operand1` and `operand2` into a sequence/list/ordered-iterable of operands